src/amp-story-player-impl.js

```diff
@@ -407,8 +411,33 @@ export class AmpStoryPlayer extends HTMLElement {
    * @private
    */
   layoutIframe_(story, iframe, visibilityState) {
-    const {href} = this.getEncodedLocation_(story.href, visibilityState);
+    const url = story.href;
+    const ampCache = this.getAttribute('amp-cache');
+
+    if (!ampCache) {
+      this.setIframeSrc_(iframe, url, visibilityState);
+      return;
+    }
+
+    if (!SUPPORTED_CACHES.includes(ampCache)) {
+      throw new Error(
+        `Unsupported cache, use one of following: ${SUPPORTED_CACHES}`
+      );
+    }
+
+    ampToolboxCacheUrl.createCacheUrl(ampCache, url).then((cacheUrl) => {
```

Comment:
(Same line different comment)
Is the fact that this method is now asynchronous impacts the rest of the code? Is it safe to not wait for this to promise to resolve?